File: Numerical Methods/C-programming/gauss jordan.cpp
#include <stdio.h>


int main() {
    int n, i, j, k, p;
    float pivot, term, a[10][11];

    printf("Enter Dimension of System of equations\n");
    scanf("%d", &n);

    printf("Enter coefficients Augmented Matrix\n");
    for (i = 0; i < n; i++) {
        for (j = 0; j < n + 1; j++) {
            scanf("%f", &a[i][j]);
        }
    }

    // Gauss-Jordan Elimination
    for (k = 0; k < n; k++) {
        pivot = a[k][k];

        // Normalization of the pivot row
        for (p = 0; p < n + 1; p++) {
            a[k][p] = a[k][p] / pivot;
        }

        // Make other rows zero
        for (i = 0; i < n; i++) {
            if (i != k) {
                term = a[i][k];
                for (j = 0; j < n + 1; j++) {
                    a[i][j] = a[i][j] - a[k][j] * term;
                }
            }
        }
    }

    // Display Solution
    printf("Solution:\n");
    for (i = 0; i < n; i++) {
        printf("x%d = %.2f\n", i + 1, a[i][n]);
    }

      getchar();
    getchar();
    return 0;
}

File: Numerical Methods/Python/gauss_jordan.py
# Write a python program for Gauss Jordan Method

def gauss_jordan():
    import numpy as np
    n = int(input("Enter the dimension of the system of equations: "))
    
    a = np.zeros((n, n+1))

    print("Enter the augmented matrix (including RHS):")
    for i in range(n):
        for j in range(n+1):
            a[i][j] = float(input(f"a[{i+1}][{j+1}]: "))

    for k in range(n):
        pivot = a[k][k]
        for j in range(n+1):
            a[k][j] /= pivot
        
        for i in range(n):
            if i != k:
                term = a[i][k]
                for j in range(n+1):
                    a[i][j] -= a[k][j] * term

    print("\nSolution:")
    for i in range(n):
        print(f"x{i+1} = {a[i][n]:.6f}")

gauss_jordan()
